tests/app/filter.test.ts

```diff
@@ -1,114 +1,85 @@
+    policy: ["Parking Maximums"],
+    expectedRange: [125, 160],
+  },
+  { desc: "land use filter", land: ["Residential"], expectedRange: [15, 22] },
+  {
+    desc: "implementation filter",
+    implementation: ["Repealed"],
+    expectedRange: [1, 2],
+  },
+  {
+    desc: "population slider",
+    populationIntervals: [4, 8],
+    expectedRange: [580, 700],
+  },
+  {
+    desc: "all cities",
+    // The other filters already enable all options by default.
+    policy: [
+      "Reduce Parking Minimums",
+      "Eliminate Parking Minimums",
+      "Parking Maximums",
+    ],
+    implementation: [
+      "Implemented",
+      "Passed",
+      "Planned",
+      "Proposed",
+      "Repealed",
+    ],
+    expectedRange: [1850, 2200],
+  },
+];
 
-  // Find and click the scope option element
-  const optionElement = await page.$(
-    `.${filterType} option:has-text("${optionText}")`
-  );
-  await optionElement.click();
-
-  const citiesAfter = await getNumCities(page);
-  const citiesDiff = Math.abs(citiesAfter - citiesBefore);
-  return citiesDiff;
-};
-
-test.fixme(
-  "scope, policy, land, implementation filter updates markers",
-  async ({ page }) => {
+for (const edgeCase of TESTS) {
+  test(`${edgeCase.desc}`, async ({ page }) => {
     await loadMap(page);
-    await openFilterPopup(page);
-    const FILTER_OPTIONS = {
-      scope: { Regional: 41 },
-      "policy-change": { "Reduce Parking Minimums": 294 },
-      "land-use": { "All Uses": 1113 },
-      "implementation-stage": { Implemented: 433 },
-    };
-    /* eslint-disable no-await-in-loop */
-    for (const filterType of Object.keys(FILTER_OPTIONS)) {
-      const optionDict = FILTER_OPTIONS[filterType];
-      const option = Object.keys(optionDict)[0];
-      const diff = await findDiff(page, filterType, option);
-      expect(diff >= optionDict[option]).toBe(true);
-    }
-    /* eslint-enable no-await-in-loop */
-  }
-);
-
-test.fixme("population slider updates markers", async ({ page }) => {
-  await loadMap(page);
-  await openFilterPopup(page);
-  const left = await page.$(".population-slider-left");
-  const right = await page.$(".population-slider-right");
-
-  // min updated on 10/10/23
-  const populationDiff = [63, 119, 565, 253, 537, 142, 124, 28, 16, 4, 0];
+    await page.locator(".filters-popup-icon").click();
 
-  /*
-    Moves slider to interval and checks to make sure the number of cities on the map is reasonable.
-
-    willIncrease is required because this function does not know if slider is the left or right slider.
-  */
-  const testSlider = async (
-    slider: ElementHandle,
-    moveToInterval: number,
-    willIncrease: boolean
-  ): Promise<void> => {
-    const citiesBefore = await getNumCities(page);
-    const sliderBefore = parseInt(await slider.inputValue(), 10);
-    await slider.fill(moveToInterval.toString());
-    const actualDiff = (await getNumCities(page)) - citiesBefore;
+    const selectIfSet = async (
```

Comment:
Can we move this outside of the for-loop so it's not being re-defined every time? Or does it not really matter?

Reply:
Sure. It doesn't matter much, and the benefit of being in the for loop is you don't have to pass `page` as an argument. But it's probably more readable to be moved.